lib/s3.persist.stream.js

```diff
@@ -0,0 +1,102 @@
+var AWS = require('aws-sdk');
+var _ = require('underscore');
+
+var stream = require('stream');
+var util = require('util');
+
+var PassThrough = stream.PassThrough;
+
+function S3PersistStream(options) {
+  var awsCredentials = new AWS.Credentials({accessKeyId: options.accessKeyId, secretAccessKey: options.secretAccessKey});
+  var s3 = new AWS.S3();
+  s3.config.credentials = awsCredentials;
+  this.s3Stream = require('s3-upload-stream')(s3);
+
+  if (!(this instanceof S3PersistStream)) {
+    return new S3PersistStream(options);
+  }
+  PassThrough.call(this, options);
+
+  if (!options.filename) {
+    throw new Error('filename must be specified for S3 persist stream');
+  }
+
+  if (!options.path) {
+    throw new Error('path must be specified for S3 persist stream');
+  }
+
+  if (!options.bucket) {
+    throw new Error('bucket must be specified for S3 persist stream');
+  }
+
+  this._filename = options.filename;
+  this._path = options.path;
+  this._bucket = options.bucket;
+  this._maxPartSize = options.maxPartSize;
+  this._metadata = options.metadata;
+
+  this._uploadParams = {
+    "Bucket": this._bucket,
+    "Key": util.format('%s/%s', this._path, this._filename),
+    "StorageClass": "REDUCED_REDUNDANCY"
```

Comment:
Not sure but feels like this should be configurable? any other classes we might want to change to in certain situations?